docs/OBJECT-LEVEL-ENCRYPTION.md

```diff
@@ -0,0 +1,84 @@
+# OBJECT LEVEL SECURITY
+
+EVE software receives its configuration from EVC. The configuration may have confidential portions, such as passwords, acces-keys etc. Even though the configuration is exchanged over a secure channel(TLS), in case of HTTPS proxies with lawful inspection, there is a chance that these sensitive information may be exposed as plain-text for a brief period of time. In the interest of protecting these sensitive objects from getting exposed during transit, we need to deploy an additional mechanism on top of TLS to protect sensitive sections of the configuration. Using object level security, one can encrypt these objects at the origin, and by using zero-knowledge key sharing mechanism, such as ECDH, one can share the secret key used for decryption at the receiving end.  Thus a proxy, without access to this secret key, will not be able to see the actual credentials/secrets etc. 
+
+
+# ECDH 
```

Comment:
All of these are done as H1 headers, which makes the document unstructured. Let's make a few H2 headers, and the details in H3/H4. We need to make it easy to see what each part is about.

Reply:
Should be taken care of